packages/enzyme/src/ShallowWrapper.js

```diff
@@ -180,6 +185,55 @@ function isPureComponent(instance) {
   return instance && instance.isPureReactComponent;
 }
 
+function getChildContext(node, hierarchy, renderer) {
+  const { instance, type: Component } = node;
+  if (!Component.childContextTypes) {
+    return {};
+  }
+  const childContext = instance.getChildContext();
+  Object.keys(childContext).forEach((key) => {
+    if (!(key in Component.childContextTypes)) {
+      const componentName = displayNameOfNode(node);
+      throw new Error(`${componentName}.getChildContext(): key "${key}" is not defined in childContextTypes.`);
+    }
+  });
+  if (typeof renderer.checkPropTypes === 'function') {
+    renderer.checkPropTypes(Component.childContextTypes, childContext, 'child context', hierarchy);
```

Comment:
are childContextTypes new in React 16?

Reply:
Not at all! I did a bit of hunting and the [first instance of `childContextTypes` I can find in the source](https://github.com/facebook/react/blob/v0.9.0/src/core/ReactCompositeComponent.js#L968-L996) is in `v0.9.0`. I believe this means it's been part of the context API since the beginning.

Reply:
Then why wouldn't every adapter, not just the 16+ ones, implement `checkPropTypes`? or, why wouldn't we implement it in enzyme-adapter-utils instead of in each adapter?

Reply:
The <16 shallow renderers call `getChildContext()` for us and implement this logic themselves. So there's no need for the <16 renderers to implement `checkPropTypes` because nothing uses them.

However, if you think the method should be there just for the sake of consistency, I can add it (and refactor it into enzyme-adapter-utils.)

Reply:
I'm confused; this problem (enzyme doesn't provide proper child context) definitely predates react 16 - since the shallow renderer only renders one level, i'm not sure how it would take child context into account?

Reply:
Gotcha. So we still have to do work for every react version, only *more* work in 16+.

Reply:
Exactly!